src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -223,9 +223,7 @@ public void StartTagHelperWritingScope(HtmlEncoder encoder)
 
             // We need to replace the ViewContext's Writer to ensure that all content (including content written
             // from HTML helpers) is redirected.
-            var buffer = new ViewBuffer(BufferScope, Path);
-            var writer = new HtmlContentWrapperTextWriter(buffer, ViewContext.Writer.Encoding);
-            ViewContext.Writer = writer;
+            ViewContext.Writer = new ViewBufferTextWriter(buffer, ViewContext.Writer.Encoding);
```

Comment:
What's the exact invariant here? That is, in our usage, what types can `ViewContext.Writer` have e.g. just `StringWriter` and `ViewBufferTextWriter`?